public/app/plugins/datasource/cloudwatch/datasource.test.ts

```diff
@@ -35,6 +35,33 @@ describe('datasource', () => {
     });
   });
 
+  describe('performTimeSeriesQuery', () => {
+    it('should return the same length of data as result', async () => {
+      const { datasource } = setup();
+      const awsRequestMock = jest.spyOn(datasource, 'awsRequest');
+      const buildCloudwatchConsoleUrlMock = jest.spyOn(datasource, 'buildCloudwatchConsoleUrl');
+      buildCloudwatchConsoleUrlMock.mockImplementation(() => '');
```

Comment:
Do we need to replace this? seems like it does not do any networking and we could actually test if the function assigns the links correctly.

Reply:
Didn't want to add it for this test. We could have another test for the links.

Reply:
No strong feelings, just felt we could easily test that if we already have this test.